### PR TITLE
fix(payment): Hide buttons that was cashed by checkout-settings 'remoteCheckoutProviders'

### PR DESCRIPTION
**Pull request by PavlenkoM**
## What/Why?
Add logic to hide payment methods that can not be processed but cashed for some time and returns in `checkoutSettings.remoteCheckoutProviders` array.

## Rollout/Rollback
Rollback this PR.

## Testing
Before:

https://github.com/user-attachments/assets/95e5e778-3fd6-4f8f-bd58-ed3f0b136ae9

After:

https://github.com/user-attachments/assets/ceed81be-2afd-4a91-a963-193f250eb08d

https://github.com/user-attachments/assets/1cdac521-55db-49f6-a859-cdb6705d8fd6

Screenshot 2026-05-27 at 19 24 38

---

> [!NOTE]
> **Low Risk**
> Checkout UI filtering only; backward-compatible optional parameter with tests and no auth or payment processing changes.
> 
> **Overview**
> Remote/wallet checkout buttons no longer appear for payment methods that are still listed in cached **`remoteCheckoutProviders`** but are marked unavailable via **`initializationData.isHidden`** on loaded payment methods.
> 
> **`getSupportedMethodIds`** now takes an optional second argument (`loadedPaymentMethods`) and excludes hidden method IDs in addition to the existing supported-method allowlist. **`CheckoutButtonContainer`** and **`CheckoutButtonList`** pass **`getPaymentMethods()`** into that helper so the UI matches live payment-method state. Unit tests cover hidden flags, missing initialization data, and edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit caf60e03a7530fb3951667f3fbcbcc4b7b8848db. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/core/src/app/customer/CheckoutButtonContainer.tsx [hunk 1]
```diff
@@ -131,8 +131,15 @@ function mapToCheckoutButtonContainerProps({
...
     const config = getConfig();
     const providers = config?.checkoutSettings.remoteCheckoutProviders ?? [];
+    const paymentMethods = checkoutState.data.getPaymentMethods() ?? [];
+    const supportedMethodIds = getSupportedMethodIds(providers);
+
+    const availableMethodIds = supportedMethodIds.filter(
+        (methodId) => !paymentMethods.find(
+            method => method.id === methodId,
+        )?.initializationData?.isHidden,
+    );
```
Comment (bc-wpietrzak) [inline packages/core/src/app/customer/CheckoutButtonContainer.tsx]: (optional) It's good as it is, but could be a bit cleaner and easier to understand, for example:

```
const hiddenMethodIds = paymentMethods
        .filter((method) => method.initializationData?.isHidden)
        .map((method) => method.id);

const availableMethodIds = getSupportedMethodIds(providers).filter(
    (methodId) => !hiddenMethodIds.has(methodId)
);
```

**Review by bc-wpietrzak — DISMISSED** [review]

File: packages/core/src/app/customer/CheckoutButtonContainer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import { createLocaleContext } from '@bigcommerce/checkout/locale';
+import { getGuestCustomer } from '@bigcommerce/checkout/test-mocks';
+import { render, screen } from '@bigcommerce/checkout/test-utils';
+
+import { getStoreConfig } from '../config/config.mock';
+
+import CheckoutButtonContainer from './CheckoutButtonContainer';
+
+jest.mock('./resolveCheckoutButton', () => {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (bc-maxy) [inline packages/core/src/app/customer/CheckoutButtonContainer.test.tsx]:
```suggestion
```
Not needed?

**Review by bc-maxy — DISMISSED** [review]

File: packages/core/src/app/customer/CheckoutButtonContainer.tsx (hunk 1 above)
Comment (PavlenkoM) [reply packages/core/src/app/customer/CheckoutButtonContainer.tsx]: thanks, you varian will be easier to read.
changed

File: packages/core/src/app/customer/CheckoutButtonContainer.test.tsx (hunk 2 above)
Comment (PavlenkoM) [reply packages/core/src/app/customer/CheckoutButtonContainer.test.tsx]: you are right, no need, thanks

**Review by bc-maxy — DISMISSED** [review]

File: packages/core/src/app/customer/CheckoutButtonContainer.tsx [hunk 3]
```diff
@@ -129,8 +129,17 @@ function mapToCheckoutButtonContainerProps({
...
+
+    const hiddenMethodIds = paymentMethods
+        .filter((method) => method.initializationData?.isHidden)
+        .map((method) => method.id);
+
+    const availableMethodIds = supportedMethodIds.filter(
+        (methodId) => !hiddenMethodIds.includes(methodId),
+    );
 
-    const availableMethodIds = getSupportedMethodIds(providers);
```
Comment (bc-peng) [inline packages/core/src/app/customer/CheckoutButtonContainer.tsx:133]: Can we put added logic into `getSupportedMethodIds` for centralising the filtering logic?

**Review by bc-wpietrzak — DISMISSED** [review]

File: packages/core/src/app/customer/CheckoutButtonContainer.tsx (hunk 3 above)
Comment (PavlenkoM) [reply packages/core/src/app/customer/CheckoutButtonContainer.tsx:133]: Good point! thanks!
Move filtering logic to `getSupportedMethodIds`

**Review by bc-wpietrzak — APPROVED** [review]

**Review by bc-peng — APPROVED** [review]

**Review by bc-maxy — APPROVED** [review]

**Comment by bc-launchbay** [comment]
*checkout-js* deployed to Staging US
success [20260528071544](https://launchbay.bigcommerce.net/projects/535/releases/286306) by Mykhailo Pavlenko

**Comment by bc-launchbay** [comment]
*checkout-js* deployed to Production US
success [20260528071544](https://launchbay.bigcommerce.net/projects/535/releases/286306) by Launchbay

**Comment by bc-launchbay** [comment]
*checkout-js* deployed to Integration US
success [20260528071544](https://launchbay.bigcommerce.net/projects/535/releases/286306) promoted by https://launchbay.bigcommerce.net/projects/535/releases/286339 being deployed. Deployed by Launchbay
